packages/flutter_tools/lib/src/base/io.dart

```diff
@@ -112,13 +111,7 @@ ExitFunction get exit {
 
 // Whether the tool is executing in a unit test.
 bool _inUnitTest() {
-  try {
-    expect(true, true);
-  } on StateError {
-    // If a StateError is caught, then this is not a unit test.
-    return false;
-  }
-  return true;
+  return Zone.current[#test.declarer] != null;
```

Comment:
This needs a test that the value is non-null in a test.

Reply:
Done